azurerm/data_source_subscription.go

```diff
@@ -0,0 +1,47 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceArmSubscription() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmSubscriptionRead,
+		Schema: map[string]*schema.Schema{
+
+			"subscription_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"display_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmSubscriptionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	groupClient := client.subscriptionsGroupClient
+
+	subscriptionId := d.Get("subscription_id").(string)
+	if subscriptionId == "" {
+		subscriptionId = client.subscriptionId
+	}
+
+	resp, err := groupClient.Get(subscriptionId)
+	if err != nil {
+		return fmt.Errorf("Error reading subscription: %s", err)
```

Comment:
given the SDK's currently masking some error messages - can we change this the formatted string to `%+v` to return the full error message?

Reply:
Done.